src/main/java/org/kuali/kfs/module/cam/document/service/impl/GlLineServiceImpl.java

```diff
@@ -408,16 +428,34 @@ protected boolean doesGeneralLedgerEntryMatchAssetAccountingDetails( GeneralLedg
         }
         
         //check if GL Debit matches acct line type target and Credit matches Source
-		if ((StringUtils.equals(entry.getTransactionDebitCreditCode(), KFSConstants.GL_DEBIT_CODE) && StringUtils.equals(accountingDetails.getFinancialDocumentLineTypeCode(), KFSConstants.TARGET_ACCT_LINE_TYPE_CODE))
-				|| (StringUtils.equals(entry.getTransactionDebitCreditCode(), KFSConstants.GL_CREDIT_CODE) && StringUtils.equals(accountingDetails.getFinancialDocumentLineTypeCode(), KFSConstants.SOURCE_ACCT_LINE_TYPE_CODE))) {
-			// this is a match, keep going
-		} else {
-			return false;
-		}     
+        boolean isGlEntryDebit = StringUtils.equals(entry.getTransactionDebitCreditCode(), KFSConstants.GL_DEBIT_CODE);
+        boolean isGlEntryCredit = StringUtils.equals(entry.getTransactionDebitCreditCode(), KFSConstants.GL_CREDIT_CODE);
+        boolean isAccountingDetailTarget = StringUtils.equals(accountingDetails.getFinancialDocumentLineTypeCode(), KFSConstants.TARGET_ACCT_LINE_TYPE_CODE);
+        boolean isAccountingDetailSource = StringUtils.equals(accountingDetails.getFinancialDocumentLineTypeCode(), KFSConstants.SOURCE_ACCT_LINE_TYPE_CODE);
+        boolean isErrorCorrection = accountingDetails.getAmount().isNegative();
```

Comment:
Another spot to verify if a negative amount is sufficient for identifying the line/details as an error correction.

Reply:
I'll add more details